src/psyclone/domain/lfric/psyir.py

```diff
@@ -0,0 +1,253 @@
+# POSSIBILITY OF SUCH DAMAGE.
+# -----------------------------------------------------------------------------
+# Author R. W. Ford STFC Daresbury Lab
+
+'''This module generates LFRic-specific PSyIR classes from lists of
+definitions.
+
+'''
+# pylint: disable=unused-import
+# pylint: disable=exec-used
+from collections import namedtuple
+from psyclone.psyir.symbols import ContainerSymbol, DataSymbol, DeferredType, \
+    GlobalInterface, ScalarType, ArrayType
+
+# Define LFRic module symbols.
+
+# The first Module named-tuple argument specifies the name of the
+# module and the second argument declares the name(s) of any symbols
+# declared by the module.
+
+Module = namedtuple('Module', ["name", "vars"])
+modules = [
+    Module("constants_mod", ["i_def", "r_def", "l_def"])]
+
+# Generate LFRic module symbols from definitions
+for module in modules:
+    MODULE_NAME = module.name
+    # Create the module (using a PSyIR ContainerSymbol)
+    exec("{0} = ContainerSymbol('{1}')\n".format(
+        MODULE_NAME.upper(), MODULE_NAME))
+    # Create the variables specified by the module (using PSyIR DataSymbols)
+    for module_var in module.vars:
+        exec("{0} = DataSymbol('{1}', DeferredType(), interface="
+             "GlobalInterface({2}))".format(
+                 module_var.upper(), module_var, MODULE_NAME.upper()))
+
+# Define generic LFRic scalar datatypes and symbols
+
+# The first Scalar named-tuple argument determines the names of the
+# resultant datatype and datasymbol classes, the second argument
+# specifies the intrinsic PSyIR type and the third argument specifies
+# the precision required by referencing symbols already declared
+# above.
+
+GenericScalar = namedtuple('GenericScalar', ["name", "intrinsic", "precision"])
+generic_scalar_datatypes = [
+    GenericScalar("lfric integer scalar", "integer", "i_def"),
+    GenericScalar("lfric real scalar", "real", "r_def"),
+    GenericScalar("lfric logical scalar", "boolean", "l_def")]
+
+# Generate generic LFRic scalar datatypes and symbols from definitions
+for info in generic_scalar_datatypes:
+    NAME = "".join(info.name.title().split())
+    INTRINSIC = info.intrinsic.upper()
+    PRECISION = info.precision
+    ARGS = ["self", "name", "interface=None"]
+    # Create the specific datatype
+    exec(
+        "class {0}DataType(ScalarType):\n"
+        "    def __init__(self):\n"
+        "        super({0}DataType, self).__init__(\n"
+        "            ScalarType.Intrinsic.{1}, {2})\n"
+        "".format(NAME, INTRINSIC, PRECISION.upper()))
+    # Create the specific symbol
+    exec(
+        "class {0}DataSymbol(DataSymbol):\n"
+        "    def __init__(self, name, interface=None):\n"
+        "        super({0}DataSymbol, self).__init__(\n"
+        "            name, {0}DataType(), interface=interface)\n"
+        "".format(NAME))
+
+# Define specific LFRic scalar datatypes and symbols
+
+# The first Scalar named-tuple argument determines the names of the
```

Comment:
The Scalar namedtype has three properties: first, the name blah blah"

Reply:
done.